src/compiler/checker.ts

```diff
@@ -17101,14 +17101,21 @@ namespace ts {
 
             // Find the first enclosing class that has the declaring classes of the protected constituents
             // of the property as base classes
-            const enclosingClass = forEachEnclosingClass(node, enclosingDeclaration => {
+            let enclosingClass = forEachEnclosingClass(node, enclosingDeclaration => {
                 const enclosingClass = <InterfaceType>getDeclaredTypeOfSymbol(getSymbolOfNode(enclosingDeclaration)!);
                 return isClassDerivedFromDeclaringClasses(enclosingClass, prop) ? enclosingClass : undefined;
             });
             // A protected property is accessible if the property is within the declaring class or classes derived from it
             if (!enclosingClass) {
-                error(errorNode, Diagnostics.Property_0_is_protected_and_only_accessible_within_class_1_and_its_subclasses, symbolToString(prop), typeToString(getDeclaringClass(prop) || type));
-                return false;
+                let thisParameter: ParameterDeclaration | undefined;
+                const thisContainer = getThisContainer(node, /* includeArrowFunctions */ false);
+                if (!thisContainer || !isFunctionLike(thisContainer) || !(thisParameter = getThisParameter(thisContainer)) || !thisParameter.type) {

```

Comment:
it is kind of weird that this works for static properties without `this.`. i think we should prohibit this. e.g.:

```ts
class C { 
    protected static S = 0;
}

C.S;  // Error

function f(this: C) { 
    C.S; // OK
}
```